pkg/util/containers/v2/metrics/containerd/collector.go

```diff
@@ -200,21 +200,13 @@ func (c *containerdCollector) refreshPIDCache(currentTime time.Time, cacheValidi
 	}
 
 	// Full refresh
-	containers, err := c.client.Containers()
+	pidsToContainers, err := c.client.PidsToContainers()
```

Comment:
I believe that this allocates one more `map[]` and a loop linear on the number of containers. As the method is not required anywhere else, I wonder if we should move the code to `util`

Reply:
Yes I was wondering the same 🤔 It's just that the `ContainerdItf` interface doesn't do any processing (for now), it just interfaces w/ the containerd AP that's why it felt a bit odd to add caching and processing there in `PidsToContainers`. No strong opinion though, I do believe that it has value to hide the complexity for the client as well in addition to saving the extra map/loop 👍